src/clj/dashboard_clj/routes.clj

```diff
@@ -23,15 +25,36 @@
   (GET "/version" _
     {:status 200
      :headers {"Content-Type" "text/json; charset=utf-8"}
-     :body (str (json/write-str {:version (version/get-version
-                                            "vanilla"
-                                            "vanilla"
-                                            "version number not found")}))})
+     :body (do
+             (prn "get version")
+             (str (json/write-str {:version (version/get-version
+                                              "vanilla"
+                                              "vanilla"
+                                              "version number not found")})))})
 
   (GET "/services" _
     {:status 200
      :headers {"Content-Type" "text/json; charset=utf-8"}
-     :body (str (json/write-str {:services (h/get-services)}))})
+     :body (do
+             (prn "get services")
+             (str (json/write-str {:services (h/get-services)})))})
+
+
+  ;TODO test removing the write-str here and put the de-string back in url handler
+  (GET "/layout" _
+    {:status 200
+     :headers {"Content-Type" "text/json; charset=utf-8"}
+     :body (do
+             (prn "get layout")
+             (str (json/write-str {:layout (h/get-layout)})))})
+
+
+  (POST "/save-layout" req
+    {:status 200
+     :headers {"Content-Type" "text/json; charset=utf-8"}
+     :body (do
+             (prn "save layout")
+             (h/save-layout (get-in req [:params :widgets])))})
```

Comment:
here is where to put the return value into a map:  {:widgets-save (h/save-layout...)}

Reply:
Handled and doesnt 500 error anymore